### PR TITLE
Add endpoint to list clients which doesn't accept a JSON list in req

### PR DESCRIPTION
**Pull request by akshaymankar**
The new endpoint is at `POST /users/list-clients/v2`, this endpoint consumes a
JSON object with one key `qualified_users` whose value must be a list of
`Qualiified UserId` limited to MaxUsersForListClientsBulk.

### CODE REVIEW

**Review by pcapriotti — APPROVED** [review]
Looks good.

File: services/brig/src/Brig/API/Public.hs [hunk 1]
```diff
@@ -328,13 +328,22 @@ type GetUserClientQualified =
     :> Get '[Servant.JSON] Public.PubClient
 
 type ListClientsBulk =
-  Summary "List all clients for a set of user ids"
+  Summary "List all clients for a set of user ids (deprecated, use /user/list-clients/v2)"
```
Comment (jschaul) [inline services/brig/src/Brig/API/Public.hs:331]:
```suggestion
  Summary "List all clients for a set of user ids (deprecated, use /users/list-clients/v2)"
```

File: services/brig/src/Brig/API/Public.hs [hunk 2]
```diff
@@ -328,13 +328,22 @@ type GetUserClientQualified =
...
     :> Post '[Servant.JSON] (Public.QualifiedUserMap (Set Public.PubClient))
 
+type ListClientsBulkV2 =
+  Summary "List all clients for a set of user ids"
+    :> ZAuthServant
+    :> "users"
+    :> "list-clients"
+    :> "v2"
+    :> Servant.ReqBody '[Servant.JSON] (Public.LimitedQualifiedUserIdList MaxUsersForListClientsBulk)
+    :> Post '[Servant.JSON] (Public.QualifiedUserMap (Set Public.PubClient))
```
Comment (jschaul) [inline services/brig/src/Brig/API/Public.hs:345]: While we're at it: is giving back a map a great idea? Wouldn't we have more flexibility for the future if we give back an object containing the map here: `{"qualifiedUserMap" : { "domain" : [ ... ] }}`? 
(feel free to ignore me here and merge the PR as is. I just thought I'd bring that idea up)

**Review by jschaul — COMMENTED** [review]
Needs an ormolu, and see inline comment/question. Otherwise fine to merge.

File: services/brig/src/Brig/API/Public.hs (hunk 2 above)
Comment (akshaymankar) [reply services/brig/src/Brig/API/Public.hs:345]: Hmm, I am going to implement the `Named` thing I did lazily, but in a slightly better way and in wire-api to achieve this.

Comment (akshaymankar) [reply services/brig/src/Brig/API/Public.hs]: Done, LMK what you think.

**Review by jschaul — APPROVED** [review]

**Pull request merged**